src/line-graph.js

```diff
@@ -119,35 +132,60 @@ export class LineGraph extends HTMLElement {
       .append('g')
       .attr("transform", `translate(${margin.top}, ${margin.left})`);
 
-
     // Create the lines
     var line = d3.line()
       .x(d => xScale(d.time))
       .y(d => yScale(d.value));
 
-    //colorScale.domain(data.map(d => d.group)); 
     // add element for line and add class name
     let lines = svg.append('g')
       .attr('class', 'lines');
-
-    // add the lines for each collection of objects to the SVG
-    lines.selectAll('.line-group')
+    
+      // create g tag with path having class line-group and line.
+      lines.selectAll('.line-group')
       .data(data).enter()
       .append('g')
       .attr('class', 'line-group')
       .append('path')
       .attr('class', 'line')
       .attr('d', d => line(d.datapoints))
       .style('stroke', d => d.color)
-      .style('fill', 'none');
-      
-    /*
-    TODO: Color schema strategy needed to ensure lines
-    are the right colors
-    https://github.com/d3/d3-scale/blob/master/README.md#sequential-scales
-    var color = d3.scaleOrdinal(d3.schemeCategory10);
-    .style('stroke', (d, i) => color(i));
-    */
+      .style('fill', 'none')
+      .each((d, i) => { // loop through datapoints to fetch time and value to create tooltip hover events with value.
+        lines.selectAll('dot')
+          .data(d.datapoints)
+          .enter()
+          .append("circle")
+          .attr("r", 4)
```

Comment:
I'm thinking this value represents the radius of the circle, I think this setting is fine. In the docs it's not quite clear where the points are but that may be because there are only 3 datapoints.

Reply:
Yes 4 is radius of circle but I have set fill value none that's why circle is not showing and also it is not required because as your second statement, there are only 6 data-points on both lines but in future once we have large data then we do not need to find. wherever we will hover, will see the data.

Reply:
But we can show datapoint as if required. this is not an issue to show!

Reply:
Yup, that's fine.